x-pack/plugins/stack_alerts/server/alert_types/es_query/executor.ts

```diff
@@ -0,0 +1,193 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { sha256 } from 'js-sha256';
+import { i18n } from '@kbn/i18n';
+import { CoreSetup, Logger } from 'kibana/server';
+import { addMessages, EsQueryAlertActionContext } from './action_context';
+import { ComparatorFns, getHumanReadableComparator } from '../lib';
+import { parseDuration } from '../../../../alerting/server';
+import { ExecutorOptions, OnlyEsQueryAlertParams, OnlySearchSourceAlertParams } from './types';
+import { ActionGroupId, ConditionMetAlertInstanceId } from './constants';
+import { fetchEsQuery } from './lib/fetch_es_query';
+import { EsQueryAlertParams } from './alert_type_params';
+import { fetchSearchSourceQuery } from './lib/fetch_search_source_query';
+import { Comparator } from '../../../common/comparator_types';
+
+export async function executor(
+  logger: Logger,
+  core: CoreSetup,
+  options: ExecutorOptions<EsQueryAlertParams>
+) {
+  const esQueryAlert = isEsQueryAlert(options);
+  const { alertId, name, services, params, state } = options;
+  const { alertFactory, scopedClusterClient, searchSourceClient } = services;
+  const currentTimestamp = new Date().toISOString();
+  const publicBaseUrl = core.http.basePath.publicBaseUrl ?? '';
+
+  const compareFn = ComparatorFns.get(params.thresholdComparator);
+  if (compareFn == null) {
```

Comment:
nit: I don't like `==` as it's not clear if it was accidental or intentional. can we expand this to be `!== null && !== undefined`

Reply:
I would keep it for now.